decorator.rb

```diff
@@ -0,0 +1,26 @@
+require './main'
+
+class Decorator < Nameable
+  # assign a nameable object from params to an instance variable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  # The Decorator delegates all work to the wrapped component.
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.strip
```

Comment:
- [ ] As in the requirement, you should have a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters
eg:
````ruby
    def correct_name
      @nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name
    end
````

Reply:
Thank you for the feedback. I will implement the changes.